### PR TITLE
Communication: Add new CallingServerClient constructor to support cre…

### PR DESCRIPTION
**Pull request by melneubert**
…ation with TokenCredential

# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [ ] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/main/eng/mgmt/mgmtmetadata).
- [ ] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running `generate.ps1`. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version.
	
	**Note: We have recently updated the PSH module called by `generate.ps1` to emit additional data. This would help reduce/eliminate the Code Verification check error. Please run following command**:

	    `dotnet msbuild eng/mgmt.proj /t:Util /p:UtilityName=InstallPsModules`

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

### CODE REVIEW

**Review by zihzhan-msft — APPROVED** [review]

File: sdk/communication/Azure.Communication.CallingServer/tests/Infrastructure/CallingServerLiveTestBase.cs [hunk 1]
```diff
@@ -115,6 +117,35 @@ protected CallingServerClient CreateInstrumentedCallingServerClient()
...
+                tokenCredential = new MockCredential();
+            }
+            else
+            {
+                #region Snippet:Azure_Communication_CallingServer_Tests_Samples_CreateCallingServerClientWithToken
+                //@@ string endpoint = "<endpoint_url>";
+                //@@ string userAssignedClientId = "<your managed identity client Id>";
+                //@@ var tokenCredential = new DefaultAzureCredential(new DefaultAzureCredentialOptions { ManagedIdentityClientId = userAssignedClientId });
+                /*@@*/
+                tokenCredential = new DefaultAzureCredential();
```
Comment (minnieliu) [inline sdk/communication/Azure.Communication.CallingServer/tests/Infrastructure/CallingServerLiveTestBase.cs:140]: line 139 and 140 should be same line. The readme snippet will also need to be updated.

File: sdk/communication/Azure.Communication.CallingServer/tests/ServerCall/ServerCallLiveTests.cs [hunk 2]
```diff
@@ -28,36 +28,39 @@ public ServerCallLiveTests(bool isAsync) : base(isAsync)
         [Test]
         public async Task RunAllRecordingFunctionsScenarioTests()
         {
-            CallingServerClient callingServerClient = CreateInstrumentedCallingServerClient();
+            CallingServerClient[] callingServerClients = new CallingServerClient[] { CreateInstrumentedCallingServerClient(), CreateInstrumentedCallingServerClientWithToken() };
```
Comment (minnieliu) [inline sdk/communication/Azure.Communication.CallingServer/tests/ServerCall/ServerCallLiveTests.cs:31]: Please consider refactoring the code to use this format to test the different authentication cases: https://github.com/Azure/azure-sdk-for-net/blob/87ac27323d7c3dd8004ef1eaad60842a1a3b0a7c/sdk/communication/Azure.Communication.Identity/tests/CommunicationIdentityClient/CommunicationIdentityClientLiveTests.cs#L32

File: sdk/communication/Azure.Communication.CallingServer/README.md [hunk 3]
```diff
@@ -35,6 +35,14 @@ var connectionString = "<connection_string>"; // Find your Communication Service
 CallingServerClient callingServerClient = new CallingServerClient(connectionString);
 ```
 
+```C# Snippet:Azure_Communication_CallingServer_Tests_Samples_CreateCallingServerClientWithToken
+string endpoint = "<endpoint_url>";
+string userAssignedClientId = "<your managed identity client Id>";
+var tokenCredential = new DefaultAzureCredential(new DefaultAzureCredentialOptions { ManagedIdentityClientId = userAssignedClientId });
```
Comment (minnieliu) [inline sdk/communication/Azure.Communication.CallingServer/README.md:41]: Why is the `ManagedIdentityClientId` required here?

File: sdk/communication/Azure.Communication.CallingServer/README.md [hunk 4]
```diff
@@ -35,6 +35,14 @@ var connectionString = "<connection_string>"; // Find your Communication Service
 CallingServerClient callingServerClient = new CallingServerClient(connectionString);
 ```
 
+```C# Snippet:Azure_Communication_CallingServer_Tests_Samples_CreateCallingServerClientWithToken
+string endpoint = "<endpoint_url>";
+string userAssignedClientId = "<your managed identity client Id>";
```
Comment (minnieliu) [inline sdk/communication/Azure.Communication.CallingServer/README.md:40]: In general, let's keep the format of the snippet as well as the place holder string names consistent with existing README: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/communication/Azure.Communication.Identity/README.md

File: sdk/communication/Azure.Communication.CallingServer/README.md [hunk 5]
```diff
@@ -35,6 +35,14 @@ var connectionString = "<connection_string>"; // Find your Communication Service
 CallingServerClient callingServerClient = new CallingServerClient(connectionString);
 ```
 
+```C# Snippet:Azure_Communication_CallingServer_Tests_Samples_CreateCallingServerClientWithToken
```
[the remark this answers is not in the document]
Comment (melneubert) [reply sdk/communication/Azure.Communication.CallingServer/README.md:39]: Fixed to match the readme you pointed me to here: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/communication/Azure.Communication.Identity/README.md

File: sdk/communication/Azure.Communication.CallingServer/README.md (hunk 4 above)
Comment (melneubert) [reply sdk/communication/Azure.Communication.CallingServer/README.md]: Done.

File: sdk/communication/Azure.Communication.CallingServer/README.md (hunk 3 above)
Comment (melneubert) [reply sdk/communication/Azure.Communication.CallingServer/README.md]: It shouldn't be there. Removed it, thanks for catching.

File: sdk/communication/Azure.Communication.CallingServer/tests/Infrastructure/CallingServerLiveTestBase.cs (hunk 1 above)
Comment (melneubert) [reply sdk/communication/Azure.Communication.CallingServer/tests/Infrastructure/CallingServerLiveTestBase.cs:135]: Updated it here

File: sdk/communication/Azure.Communication.CallingServer/tests/ServerCall/ServerCallLiveTests.cs (hunk 2 above)
Comment (melneubert) [reply sdk/communication/Azure.Communication.CallingServer/tests/ServerCall/ServerCallLiveTests.cs]: Thanks for the suggestion. Updated it here.

File: sdk/communication/Azure.Communication.CallingServer/README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (melneubert) [reply sdk/communication/Azure.Communication.CallingServer/README.md:39]: Added it to match similar snippet here: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/communication/Azure.Communication.Identity/README.md

**Review by minnieliu — APPROVED** [review]
LGTM! Thanks for addressing all the feedback

**Pull request merged**